Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -6,126 +6,136 @@
 //
 
 import Foundation
-
-/// `List<ModelType>` is a DataStore-aware custom `Collection` that is capable of loading
-/// records from the `DataStore` on-demand. This is specially useful when dealing with
-/// Model associations that need to be lazy loaded.
-///
-/// When using `DataStore.query(_ modelType:)` some models might contain associations
-/// with other models and those aren't fetched automatically. This collection keeps track
-/// of the associated `id` and `field` and fetches the associated data on demand.
-public class List<ModelType: Model>: Collection, Codable, ExpressibleByArrayLiteral {
-
-    public typealias Index = Int
+import Combine
+
+/// `List<ModelType>` provides simple conformance to `Collection` with a backing array of `Model` type elements.
+/// This class acts as an abstract class for plugins to build subclasses that implement their own specific
+/// implementations of a `ModelList`. The decoding logic leverages the `ModelListRegistry` to check for decodability
+/// and decodes to subclasses of this type.
+open class List<ModelType: Model>: ModelList {
+    @available(iOS 13.0, *)
+    public typealias LazyListPublisher = AnyPublisher<Elements, DataStoreError>
+
+    public typealias ModelListElement = ModelType
+    public typealias Page = List<ModelType>
     public typealias Element = ModelType
     public typealias Elements = [Element]
 
-    public typealias ArrayLiteralElement = ModelType
-
     /// The array of `Element` that backs the custom collection implementation.
-    internal var elements: Elements
-
-    /// If the list represents an association between two models, the `associatedId` will
-    /// hold the information necessary to query the associated elements (e.g. comments of a post)
-    internal var associatedId: Model.Identifier?
-
-    /// The associatedField represents the field to which the owner of the `List` is linked to.
-    /// For example, if `Post.comments` is associated with `Comment.post` the `List<Comment>`
-    /// of `Post` will have a reference to the `post` field in `Comment`.
-    internal var associatedField: ModelField?
-
-    internal var limit: Int = 100
-
-    /// The current state of lazily loaded list
-    internal var state: LoadState = .pending
+    public var elements: Elements
```

Comment:
this has been made public as part of providing developers access to the backing `[]`